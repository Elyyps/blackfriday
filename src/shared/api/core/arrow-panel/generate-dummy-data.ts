import { IArrowPanelProps } from "@app/core/arrow-panel";
import party from "@assets/icons/party.svg";
import iears from "@assets/dummy-images/product-thumbnails/iears.png";
import imac from "@assets/dummy-images/product-thumbnails/imac.png";
import iphone from "@assets/dummy-images/product-thumbnails/iphone.png";
import loader from "@assets/dummy-images/product-thumbnails/loader.png";
import macbook from "@assets/dummy-images/product-thumbnails/macbook.png";
import mouse from "@assets/dummy-images/product-thumbnails/mouse.png";

export const generateArrowPanelData = (): IArrowPanelProps[] => {
  const data: IArrowPanelProps[] = [
    {
      image: iears,
      link: { url: "/", title: "Apple Airpods" }
    },
    {
      image: imac,
      link: { url: "/", title: "Apple iPhones" }
    },
    {
      image: iphone,
      link: { url: "/", title: "Apple MacBooks" }
    },
    {
      image: loader,
      link: { url: "/", title: "Apple iMacs" }
    },
    {
      image: macbook,
      link: { url: "/", title: "Categorie item met een lange naam" }
    },
    {
      image: mouse,
      link: { url: "/", title: "Categorie item met een nog langere naam" }
    },
    {
      link: { url: "/", title: "Categorie item zonder afbeelding" }
    },
    {
      icon: party,
      link: { url: "/", title: "Categorie met icoon" }
    }
  ];

  return data;
};
