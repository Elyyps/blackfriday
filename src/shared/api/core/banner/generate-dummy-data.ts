import { Banner } from "./banner";
import CoolBlue from "@assets/cool-blue.svg";

export const generateDummyBannerComponentData = (): Banner => ({
  title: "Black Friday bij Coolblue",
  buttonTitle: "Naar deals",
  buttonLink: "/deals-overview",
  image: {
    src: "https://viper-development-images.s3-eu-west-1.amazonaws.com/blackFriday/ad.png",
    alt: "Ad picture"
  },
  imageLink: "https://coolblue.nl",
  label: "22 t/m 26 nov.",
  logo: { src: CoolBlue, alt: "CoolBlue logo" },
  text: "Een selectie van bizarre aanbiedingen.",
  moreInfoLink: "https://coolblue.nl",
  showAlternativeBanner: false
});
