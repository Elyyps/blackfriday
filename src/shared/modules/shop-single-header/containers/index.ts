export * from "./shop-single-header-container";
