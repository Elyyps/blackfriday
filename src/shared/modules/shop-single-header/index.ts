export * from './shop-single-header.component'