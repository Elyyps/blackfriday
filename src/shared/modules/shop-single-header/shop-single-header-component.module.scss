.product-detail {
  box-sizing: content-box;
  background-color: $bg-muted;
  max-width: 1200px;
  margin-left: auto;
  margin-right: auto;
  padding: 40px 32px 72px 32px;
  @media (max-width: $breakpoint-xsmall-max) {
    padding: 0;
    padding-top: 16px;
    padding-bottom: 48px;
    .card-wrap {
      padding: 0 16px;
      margin-bottom: 16px;
    }
  }
  .uk-margin-small-top {
    margin-top: 8px;
    @media (max-width: $breakpoint-small-max) {
      margin-top: 0;
    }
  }
  .card {
    &-shop {
      height: auto;
      padding-bottom: 52px;
      @media (max-width: $breakpoint-small-max) {
        padding-bottom: 16px;
        margin-bottom: 16px;
      }
    }
    &-message {
      padding: 32px;
      display: flex;
      flex-direction: column;
      align-items: center;
      color: $global-inverse-color;
      background-color: $gray-dark;
      text-align: center;
      position: relative;
      &:after {
        content: "";
        width: 40px;
        height: 40px;
        background-color: $gray-dark;
        transform: rotate(45deg) translateX(-50%);
        position: absolute;
        top: 0;
        left: 50%;
        @media (max-width: $breakpoint-xsmall-max) {
          height: 20px;
          width: 20px;
        }
      }
      .card {
        &__body {
          margin-top: 18px;
          overflow: hidden;
        }
        &__action {
          margin-top: 16px;
          .button {
            width: 100%;
          }
        }
        &__head {
          .icon-svg {
            path {
              fill: $global-inverse-color;
            }
          }
        }
      }
    }
  }
  &__list {
    display: flex;
    flex-direction: column;
    margin-bottom: 48px;
    &__item {
      svg {
        path {
          stroke: #00a82f;
        }
      }
      align-items: center;
      display: inline-flex;
      margin: 6px 0;
      font-family: Assistant;
      font-size: 16px;
      font-weight: 600;
      font-style: normal;
      font-stretch: normal;
      line-height: 1.29;
      letter-spacing: normal;
      color: #777777;
      &__text {
        margin-left: 16px;
      }
    }
  }
  &__header {
    height: 216px;
    overflow: hidden;
    border-radius: 2px 2px 0px 0px;
    img {
      object-fit: cover;
      width: 100%;
      height: 100%;
    }
  }
  &__body {
    background-color: $global-inverse-color;
    border-radius: 2px;
  }
  &__content {
    padding: 16px 32px 32px;
    &__subtitle {
      display: flex;
      align-items: flex-start;
      margin-bottom: -8px;
    }
    @media (max-width: $breakpoint-xsmall-max) {
      padding: 16px;
      padding-bottom: 48px;
    }
    h1 {
      font-size: 32px;
      font-weight: bold;
      font-style: normal;
      font-stretch: normal;
      line-height: normal;
      letter-spacing: normal;
    }
    .breadcrumbs {
      display: flex;
      margin: 0;
      padding: 0;
      list-style: none;
      margin: 0 -10px;
      font-size: 12px;
      li {
        padding: 0 10px;
        display: flex;
        font-weight: 600;
        justify-content: space-between;
        color: #999999;
        position: relative;
        > a {
          color: inherit;
          text-decoration: none;
        }
        &:not(:last-child) {
          &:after {
            content: "";
            width: 10px;
            height: 1px;
            color: #999999;
            background-color: #999999;
            position: absolute;
            top: 50%;
            right: -4px;
          }
        }
      }
    }
    .content {
      &__see-more {
        display: block;
        width: 70px;
        margin-bottom: 16px;
        margin-top: -36px;
      }
      position: relative;
      &__gradient {
        background: -moz-linear-gradient(
          top,
          rgba(255, 255, 255, 0) 0%,
          rgba(255, 255, 255, 1) 76%,
          rgba(255, 255, 255, 1) 100%
        ); /* FF3.6-15 */
        background: -webkit-linear-gradient(
          top,
          rgba(255, 255, 255, 0) 0%,
          rgba(255, 255, 255, 1) 76%,
          rgba(255, 255, 255, 1) 100%
        ); /* Chrome10-25,Safari5.1-6 */
        background: linear-gradient(
          to bottom,
          rgba(255, 255, 255, 0) 0%,
          rgba(255, 255, 255, 1) 76%,
          rgba(255, 255, 255, 1) 100%
        ); /* W3C, IE10+, FF16+, Chrome26+, Opera12+, Safari7+ */
        filter: progid:DXImageTransform.Microsoft.gradient( startColorstr='#00ffffff', endColorstr='#ffffff',GradientType=0 ); /* IE6-9 */
        width: 100%;
        position: absolute;
        bottom: 0;
        left: 0;
        right: 0;
        top: calc(100% - 80px);
      }
      &__head {
        display: flex;
        align-items: center;
        margin-bottom: 24px;
        justify-content: space-between;
        &__stars {
          font-size: 12px;
          font-weight: bold;
          font-style: normal;
          font-stretch: normal;
          line-height: normal;
          letter-spacing: normal;
          color: #00a82f;
          display: flex;
          flex-direction: column;
          align-items: flex-end;
          text-align: right;
          white-space: nowrap;
        }
      }
      &__body {
        margin: 32px 0;
        overflow: hidden;
      }
    }
    .uk-list {
      margin-bottom: 32px;
    }
  }
  &__footer {
    border-top: 1px solid $tint-gray-light;
    padding-top: 16px;
    .actions {
      display: flex;
      flex-direction: row;
      align-items: center;
      flex-wrap: wrap;
      justify-content: space-between;
      margin-bottom: 8px;
      @media (max-width: $breakpoint-xsmall-max) {
        margin-bottom: 0;
      }

      &__item {
        display: flex;
        align-items: center;
        order: 1;
        margin-bottom: 16px;
        @media (max-width: $breakpoint-xsmall-max) {
          flex: 1 1 100%;
          justify-content: space-between;
          & + .actions__item {
            order: 0;
            margin-bottom: 16px;
          }
        }

        &-wrap {
          margin: 0 -4px;
          flex: 1 1 100%;
          display: flex;
          > * {
            margin: 0 4px;
            @media (max-width: $breakpoint-xsmall-max) {
              width: 50%;
            }
          }
        }
        .button {
          &--link {
            padding: 0;
            @media (max-width: $breakpoint-xsmall-max) {
              width: auto;
            }
          }
        }
      }
      &__item__first {
        display: inline-flex;
        align-items: baseline;
        :hover {
          color: $brand_secondary !important;
          svg {
            path {
              fill: $brand_secondary !important;
            }
          }
        }
        svg {
          margin-left: 8px;
          path {
            fill: #777;
          }
        }
      }
    }
    .labels {
      display: flex;
      align-items: center;
      > * {
        margin: 0 4px;
        border-radius: 2px;
      }
      &:first-child {
        margin-left: 0px;
      }
      &__item {
        padding: 0 4px;
      }
    }
  }
  .rating {
    flex-direction: column;
    align-items: flex-end;
    &__list {
      margin: 0;
      padding: 0;
      list-style: none;
      display: flex;
      margin: 0 -2px;
      > li {
        padding: 0 2px;
      }
      .icon-svg {
        svg {
          fill: $tint-gray;
        }
      }
    }
    &__description {
      font-size: 12px;
      font-weight: bold;
      color: #00a82f;
      margin-top: 4px;
    }
  }
}

.container {
  box-sizing: content-box;
  background-color: $bg-muted;
  width: 100%;
  height: 100%;
}
