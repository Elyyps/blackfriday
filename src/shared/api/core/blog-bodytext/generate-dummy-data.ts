import { BlogBodyTextModule } from "./blog-bodytext";
import { generateSidebarData } from "../sidebar/generate-dummy-data";

export const generateDummyBlogBodyText = (): BlogBodyTextModule => ({
  bodyText: {
    html: `<h2>
        Hogere kortingen, meer producten en meer uiteenlopende productcategorieën. Dit gaat er gebeuren met Black
        Friday.
      </h2>
      <h3>1. Wees er snel bij, winkels beginnen eerder met Black Friday</h3>
      <p>
        Vorig jaar zagen we deze trend al opkomen. Winkels willen altijd graag de eerste zijn met hun deal, daarom
        beginnen een aantal winkels niet op Black Friday, maar vaak een paar dagen eerder. Het is daarom belangrijk dat
        je niet alleen op 23 november 2018 <a href="#">Black Friday Nederland</a> goed in de gaten houdt, maar ook
        in de week voor de kortingsdag van het jaar de website al iedere dag even checkt. Je weet nooit wanneer de deals
        online komen. Toch zien we wel dat de allerbeste deals pas echt vanaf Black Friday ingaan, maar wees er dan snel
        bij! Want voordat je het weet is jouw favoriete product uitverkocht.
      </p>
      <h3>2. Groots uitpakken met Telecom deals</h3>
      <p>
        Telecom deals zijn niet meer weg te denken van Black Friday. Wij verwachten dat er dit jaar nog meer Telecom
        deals online komen dan vorig jaar. Ben je dus op zoek naar een <a href="#">mobiel</a>? Wacht dan nog even
        tot Black Friday want we verwachten dat providers als <a href="#">Vodafone</a>, <a href="#">T-Mobile</a>
        , <a href="#">BEN</a> en <a href="#">Tele2</a> weer groots uit gaan pakken met de smartphone deals van
        2018. De Samsung Galaxy of de Apple Iphone zijn smartphones waar je altijd wel een fikse korting op kunt vinden.
        Kun je niet wachten? Bekijk dan alvast de <a href="#">Samsung Galaxy deals</a> van dit moment.
      </p>
      <h3>3. 45% meer deelnemende winkels in 2019</h3>
      <img src="https://viper-development-images.s3-eu-west-1.amazonaws.com/blackFriday/blog/3.png" width="279px" height="146px" className="align-right" alt="image description" />
      <p>
        Naar verwachting doen er in 2018 bijna de helft meer winkels mee aan <a href="#">Black Friday 2018</a>.
        Daardoor zal het aantal deals ook bijna met de helft stijgen. Het is dus belangrijk dat je jouw favoriete winkel
        goed in de gaten houdt, want voordat je het weet wordt je ondergesneeuwd door de kortingen en kun je jouw deal
        niet meer vinden.
      </p>
      <h3>4. Meer lokale scherpe aanbiedingen</h3>
      <p>
        Behalve grote namen als <a href="#">adidas</a>, <a href="#">MediaMarkt</a>, <a href="#">Ziggo</a> en
        <a href="#">Bol.com</a> is het ook slim om dit jaar de kleine lokale winkels goed in de gaten te houden.
        Naar verwachting zullen de kleine winkels en boetiekjes in jouw dorp of stad ook meedoen aan Black Friday. Black
        Friday Nederland richt zich echter alleen op de landelijke bekende winkels, dus de aanbiedingen van dat kleine
        winkeltje in jouw dorp zul je zelf in de gaten moeten houden. Dit kan vaak al door de winkel te volgen op Social
        Media en even te vragen of ze dit jaar meedoen aan Black Friday.
      </p>
      <h3>5. Meer producten in reis-, interieur- en fashionbranche</h3>
      <p>
        Dit jaar verwachten wij dat vooral de reisbranche uit gaat pakken. Deze speler was er vorig jaar ook bij met
        onder andere merken als: <a href="#">Ryanair</a>, Hotels.com, <a href="#">Expedia</a> en Hilton. De
        verwachting is vooral dat er kortingen komen op vliegtickets en hotels met Black Friday 2018. Houd onze
        reispagina goed in de gaten voor een recent overzicht van alle <a href="#">reisaanbiedingen</a> met Black
        Friday.
      </p>`,
    bottomPadding: { mobilePadding: 32, desktopPadding: 72 },
    topPadding: { mobilePadding: 32, desktopPadding: 72 },
    containerAlignment: "center",
    containerSize: "full",
    bottomMargin: "",
    topMargin: "",
    id: "1",
    name: "BlogBodyText"
  },
  dateTime: "2 minuten leestijd",
  sidebarItems: generateSidebarData(),
  id: "1",
  name: "BlogBodyTextModule",
  bottomMargin: "",
  topMargin: ""
});

export const generateBlackFriday2019Text = (): BlogBodyTextModule => ({
  bodyText: {
    html: `<p><em>Profiteren van korting tijdens <strong>Black Friday 2019?!</strong> Hier lees je alles over de achtergronden van deze Amerikaanse traditie én hoe je er dit jaar zelf van profiteert.</em></p>
<p><strong><a href="#">Black Friday</a> is een Amerikaanse feestdag die altijd wordt gevierd op de dag na <a href="#">Thanksgiving</a>. De naam is ontstaan vanwege het feit dat het altijd heel erg druk was in de stad op de dag na Thanksgiving want iedereen was vrij. De dag na Thanksgiving was dus dé perfecte dag om koopjes te scoren en met scherpe deals voor de dag te komen. Nederland heeft deze traditie overgenomen en daardoor is het op de vierde vrijdag in november altijd gekkenhuis in de winkelstraten.</strong></p>
<p><img class="aligncenter wp-image-10314 size-full" src="https://www.blackfridaynederland.nl/wp-content/uploads/2019/05/Banner-black-friday-2019.png" alt="black friday 2019" width="750" height="250" srcset="https://www.blackfridaynederland.nl/wp-content/uploads/2019/05/Banner-black-friday-2019.png 750w, https://www.blackfridaynederland.nl/wp-content/uploads/2019/05/Banner-black-friday-2019-300x100.png 300w" sizes="(max-width: 750px) 100vw, 750px"></p>
<h2><strong>Black Friday in Nederland 2019</strong></h2>
<p>Black Friday 2019 valt dit jaar op <strong>29 november.</strong> Op deze dag stunten ook in Nederland weer een groot aantal (web)winkels met dé beste aanbiedingen. Bij de start van <a href="#">Black Friday Nederland</a> in 2015 waren er maar liefs 35 (web)winkels met Black Friday aanbiedingen in Nederland maar afgelopen jaar is dit aantal gestegen tot 166 deelnemers. We verwachten dat dit aantal in 2019 alleen maar meer toe gaat nemen.</p>
<h3><strong>Verwachtingen voor Black Friday 2019</strong></h3>
<p>Vorig jaar selecteerde Black Friday Nederland maar liefst meer dan 166 Nederlandse winkels met scherpe deals. Er waren veel goede aanbiedingen te vinden in onder andere de&nbsp;<a href="#">elektronica</a> en <a href="#">fashion</a>&nbsp;branche, maar ook de <a href="#">gaming</a>, <a href="#">speelgoed</a> en <a href="#">vrije tijd</a> branche hadden scherpe deals die je zeker niet wilde missen. Dit keer verwachten we dat Black Friday zeker weer veel mooie deals gaat brengen en het nog groter gaat worden dan het al is.</p>
<p>Denk bijvoorbeeld aan lokale, kleine ondernemers in je eigen winkelstraat die toch ook mee gaan doen aan de rage, of de top 5 aan (web)winkels die allemaal dé beste deal willen hebben. Het wordt dus heel belangrijk dat je scherp in de gaten houdt wat voor deal jij tijdens Black Friday wilt scoren want voordat je het weet is jouw koopje al&nbsp; verdwenen!</p>
<h3><strong>Grootste spelers tijdens Black Friday</strong></h3>
<p>Er zijn een aantal winkels bij wie je er nu al vanuit kunt gaan dat ze volgend jaar weer van de partij zijn. <a href="#">MediaMarkt</a> heeft bijvoorbeeld elk jaar met Black Friday scherpe elektronica aanbiedingen. Hetzelfde geldt&nbsp; voor <a href="#">Coolblue</a>. Huishoudelijke apparatuur zoals wasmachines, stofzuigers of <a href="#">elektronica</a> artikelen als laptops en televisies zijn ineens een stuk beter betaalbaar met Black Friday bij Coolblue. Verder is <a href="#">Bol.com</a> ook een grote speler op de retailermarkt waar je een paar weken voor Sinterklaas- en kerst goedkoop al je cadeautjes kunt bestellen. Behalve deze drie deelnemers zijn er natuurlijk nog honderden andere deelnemende winkels.</p>
<h3><strong>Profiteren van Black Friday 2019 – Alle deals op een rijtje</strong></h3>
<p>Om orde te creëren in de chaos van alle aanbiedingen tijdens Black Friday 2019, zetten wij de beste deals voor je in een handig overzicht. Dankzij onze contacten met alle (web)winkels zijn wij als eerste op de hoogte van de beste aanbiedingen tijdens Black Friday. Wij updaten regelmatig het winkeloverzicht met de meest scherpe deals zodat jij als eerste weet wat de beste aanbiedingen worden tijdens Black Friday. Check regelmatig het winkeloverzicht voor het meest recente overzicht zodat jij zeker weet dat er geen enkele deal aan je neus voorbijgaat. Abonneer je verder op de Deal Alerts. Door je te abonneren op deze nieuwsbrief ontvang je voorafgaand en tijdens Black Friday regelmatig alerts met de beste deals. Deze deals kun jij vervolgens helemaal zelf afstemmen op je interesses.</p>`,
    bottomPadding: { mobilePadding: 32, desktopPadding: 72 },
    topPadding: { mobilePadding: 32, desktopPadding: 72 },
    containerAlignment: "center",
    containerSize: "full",
    bottomMargin: "",
    topMargin: "",
    id: "1",
    name: "BlogBodyText"
  },
  dateTime: "",
  sidebarItems: generateSidebarData(),
  id: "1",
  name: "BlogBodyTextModule",
  bottomMargin: "48px",
  topMargin: ""
});
