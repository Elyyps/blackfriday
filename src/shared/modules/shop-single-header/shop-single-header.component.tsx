/* tslint:disable */
import * as React from "react";
import styles from "./shop-single-header-component.module.scss";
import IconDefault from "@assets/icons/link.svg";
import CheckIcon from "@assets/icons/check.svg";
import HeartIcon from "@assets/icons/heart-filled.svg";
import { IconComponent } from "@app/prep/modules-prep/core";
import { DiscountCardComponent } from "@app/core/discount-card";
import { NewsletterComponent } from "@app/core/newsletter";
import { StarsRatingComponent } from "@app/core/stars-rating";
import { KeywordTagComponent } from "@app/core/keyword-tag";
import { Button } from "@app/core";
import { BackLinkComponent } from "@app/core/back-button";
import { ShopSingleHeaderModule } from "@app/api/modules/shop-single-header/shop-single-header.module";
import { generateDummyNewsletterModule } from "@app/api/modules/newsletter/generate-dummy-data";
import { ShareSocialDropdownComponent } from "@app/core/share-social-dropdown";
import { socialMediaButtons } from "@app/api/core/share-social/generate-dummy-data";
import { LinkComponent } from "@app/core/link";
import { BodyTextComponent } from "@app/core/bodytext";
import { injectIntl, InjectedIntlProps } from "react-intl";
import { ClickableComponent } from "@app/core/clickable";
import { BreadcrumbComponent } from "@app/core/breadcrumb";
import { ShadowCardComponent } from "@app/core/shadow-card";

export interface IShopSingleHeaderComponentProps {
  shopSingleHeaderModule: ShopSingleHeaderModule;
}

const component = (props: IShopSingleHeaderComponentProps & InjectedIntlProps) => {
  const {
    breadcrumbProps,
    checkList,
    CheckListTitle,
    bodyTextModule,
    couponCode,
    discountButton,
    discountPicture,
    discountSubtitle,
    discountTitle,
    favoriteButton,
    keywords,
    picture,
    rating,
    smallBackLink,
    storeLink,
    timeLeftBar,
    title,
    shareSocial
  } = props.shopSingleHeaderModule;

  return (
    <div className={styles["container"]}>
      <div className={styles["product-detail"]}>
        <div className={"uk-grid uk-grid-small"}>
          <div className={"uk-width-1-3@s"}>
            <div className={styles["card-wrap"]}>
              <React.Fragment>
                <DiscountCardComponent
                  status={timeLeftBar.text}
                  buttonText={discountButton.title}
                  buttonLink={discountButton.url}
                  content={discountSubtitle}
                  coupon={couponCode}
                  image={discountPicture}
                  title={discountTitle}
                  range={timeLeftBar.value}
                />
                <div className={"uk-visible@s"}>
                  <NewsletterComponent newsletterModule={generateDummyNewsletterModule()} />
                </div>
              </React.Fragment>
            </div>
          </div>
          <div className={"uk-width-2-3@s"}>
            <ShadowCardComponent>
              <div className={styles["product-detail__body"]}>
                <div className={styles["product-detail__header"]}>
                  <img src={picture} alt="image" />
                </div>
                <div className={styles["product-detail__content"]}>
                  {breadcrumbProps && (
                    <div className={styles["product-detail__content__subtitle"]}>
                      <BreadcrumbComponent links={breadcrumbProps.links} backButton={breadcrumbProps.backButton} />
                    </div>
                  )}
                  <div className={styles["content"]}>
                    <div className={styles["content__head"]}>
                      <h1 style={{ margin: "0" }}>{title}</h1>
                      <div className={`${styles["content__head__stars"]} uk-visible@s`}>
                        <StarsRatingComponent rating={rating.value} voteTrigger={() => {}} />
                        <span>{rating.text}</span>
                      </div>
                    </div>
                    <div className={styles["content__body"]}>
                      <BodyTextComponent style={{ margin: 0, padding: 0 }} bodyTextModule={bodyTextModule} />
                    </div>
                    {checkList && (
                      <div className="uk-visible@s">
                        <h4>{CheckListTitle}</h4>
                        <div className={styles["product-detail__list"]}>
                          {checkList.map((item, key) => (
                            <div key={key} className={styles["product-detail__list__item"]}>
                              <IconComponent size={"16px"} icon={CheckIcon} />
                              <div className={styles["product-detail__list__item__text"]}>{item}</div>
                            </div>
                          ))}
                        </div>
                      </div>
                    )}
                  </div>
                  <div className={styles["product-detail__footer"]}>
                    <div className={styles["actions"]} data-uk-margin>
                      <div className={styles["actions__item"]}>
                        <ClickableComponent
                          variant="link-primary"
                          href={storeLink.url}
                          title={storeLink.title}
                          iconStyle="filled"
                          icon={IconDefault}
                        />

                        <div className={`${styles["content__head__stars"]} uk-hidden@s`}>
                          <StarsRatingComponent rating={rating.value} voteTrigger={() => {}} />
                          {rating.text}
                        </div>
                      </div>
                      <div className={styles["actions__item"]}>
                        <div className={styles["actions__item-wrap"]}>
                          <div className="uk-hidden@s">
                            <Button
                              icon={HeartIcon}
                              iconPosition="left"
                              title={smallBackLink.title}
                              variant={"secondary"}
                            />
                          </div>
                          <ShareSocialDropdownComponent
                            buttonTitle={props.intl.formatMessage({ id: "social-media-share" })}
                            shareSocial={shareSocial}
                          />
                        </div>
                      </div>
                    </div>
                    <div className={`${styles["labels"]} uk-visible@s`}>
                      {keywords.map((item, key) => (
                        <KeywordTagComponent style={{ marginLeft: key == 0 ? "0" : "4px" }} key={key}>
                          {item}
                        </KeywordTagComponent>
                      ))}
                    </div>
                  </div>
                  <div className="uk-hidden@s">
                    <NewsletterComponent newsletterModule={generateDummyNewsletterModule()} />
                  </div>
                </div>
              </div>
            </ShadowCardComponent>
          </div>
        </div>
      </div>
    </div>
  );
};

const ShopSingleHeaderComponent = injectIntl(component);
export { ShopSingleHeaderComponent };
