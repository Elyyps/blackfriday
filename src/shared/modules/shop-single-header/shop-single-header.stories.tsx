import * as React from "react";

import { storiesOf } from "@storybook/react";
import { withA11y } from "@storybook/addon-a11y";
import { withProvider } from "@app/util";
import { ShopSingleHeaderComponent } from "./shop-single-header.component";
import { generatedummyShopSingleHeaderData } from "@app/api/modules/shop-single-header/generate-dummy-data";
import { ViewType } from "@app/stores/settings";

storiesOf("Shop Single Header", module)
  .addDecorator(withA11y)
  .addDecorator(withProvider)
  .add("Basic implementation", () => (
    <ShopSingleHeaderComponent
      screenSize={{ breakpointPixels: 0, viewType: ViewType.Desktop }}
      shopSingleHeaderModule={generatedummyShopSingleHeaderData()}
    />
  ));
