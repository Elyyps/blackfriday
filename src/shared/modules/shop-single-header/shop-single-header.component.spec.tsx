import React from "react";
import { shallow } from "enzyme";
import { ShopSingleHeaderComponent } from "./shop-single-header.component";
import { generatedummyShopSingleHeaderData } from "@app/api/modules/shop-single-header/generate-dummy-data";

describe("[ShopSingleHeader]", () => {
  it("should render component without crashing", () => {
    const renderedComponent = shallow(
      <ShopSingleHeaderComponent shopSingleHeaderModule={generatedummyShopSingleHeaderData()} />
    );
    expect(renderedComponent).toMatchSnapshot();
  });
});
