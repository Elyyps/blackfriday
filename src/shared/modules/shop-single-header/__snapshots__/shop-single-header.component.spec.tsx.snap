// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[ShopSingleHeader] should render component without crashing 1`] = `
<component
  intl={
    Object {
      "defaultFormats": Object {},
      "defaultLocale": "en",
      "formatDate": [Function],
      "formatHTMLMessage": [Function],
      "formatMessage": [Function],
      "formatNumber": [Function],
      "formatPlural": [Function],
      "formatRelative": [Function],
      "formatTime": [Function],
      "formats": Object {},
      "formatters": Object {
        "getDateTimeFormat": [Function],
        "getMessageFormat": [Function],
        "getNumberFormat": [Function],
        "getPluralFormat": [Function],
        "getRelativeFormat": [Function],
      },
      "locale": "en",
      "messages": Object {},
      "now": [Function],
      "onError": [Function],
      "textComponent": "span",
      "timeZone": null,
    }
  }
  shopSingleHeaderModule={
    Object {
      "CheckListTitle": "De voordelen van shoppen bij Beddenreus tijdens Black Friday.",
      "backLink": Array [
        Object {
          "title": "Back",
          "url": "/",
        },
      ],
      "bodyTextModule": Object {
        "containerAlignment": "left",
        "containerSize": "full",
        "html": " <div>
    <p>
      Sed ipsum metus, sodales sit amet vulputate a, lobortis mollis odio. Orci varius natoque penatibus et magnis dis
      parturient montes, nascetur ridiculus mus. Vivamus sed velit in ligula finibus viverra. Donec eleifend commodo
      ipsum nec commodo. Cras vehicula lectus et felis suscipit, vel commodo metus sagittis. Vestibulum quam elit,
      dignissim sit amet risus in, luctus accumsan lectus. Sed rutrum metus ut massa tempus, eget blandit lacus
      molestie. Pellentesque sit amet aliquam lacus. Etiam ultrices velit eu neque pharetra, vitae scelerisque sapien
      fermentum. Morbi egestas bibendum est. Donec at felis hendrerit, iaculis lorem vel, porta enim. Quisque suscipit
      condimentum elementum. Nam malesuada ut ligula sit amet porta. Sed nec gravida velit, eget porta metus. Quisque
      tempor commodo pulvinar. Nullam sagittis posuere mauris, in interdum tellus vulputate id.
    </p>
     <h1>
      <strong>Black Friday Beddenreus</strong>
    </h1>
     <p>
      Noteer vrijdag 29 november alvast in uw agenda, want dan start Black Friday 2019. Vanaf dat moment kunt u mooie
      deals en kortingen scoren.Hier in België hebben we het geluk dat de aanbiedingen in veel winkels meerdere dagen
      geldig zijn en dat we dus niet afhankelijk zijn van enkel en alleen maar één dag.Hierdoor heeft u meer tijd om al
      uw wensen bij elkaar te shoppen.
    </p>
  </div>",
        "id": "",
        "name": "shopSingleBodyText",
      },
      "breadcrumbProps": Object {
        "backButton": Object {
          "title": "Terug",
          "url": "/#",
        },
        "links": Array [
          Object {
            "title": "Home",
            "url": "/",
          },
          Object {
            "title": "Alle winkels",
            "url": "/stores-overview",
          },
        ],
        "variant": "dark",
      },
      "checkList": Array [
        "Profiteer van 15% op korting op alles",
        "Online én in de winkels",
        "De hele week aanbiedingen van 16-11 t/m 26-11",
        "Ruimere openingstijden",
      ],
      "couponCode": "black15",
      "discountButton": Object {
        "title": "Naar deals",
        "url": "/",
      },
      "discountPicture": "cool-blue.svg",
      "discountSubtitle": "Gebruik deze kortingscode:",
      "discountTitle": "15% korting op het hele assortiment",
      "favoriteButton": Object {
        "title": "Winkel opslaan als favoriet",
        "url": "/",
      },
      "id": "1",
      "keywords": Array [
        "Wonen & slapen",
        "Swiss sense",
        "hastens",
      ],
      "name": "ShopSingleHeaderModule",
      "picture": "https://viper-development-images.s3-eu-west-1.amazonaws.com/blackFriday/stores-single.png",
      "rating": Object {
        "text": "Beoordeel deze deal",
        "value": 5,
      },
      "shareSocial": Object {
        "email": Object {
          "nameSocialShare": "Email",
          "subject": "BlackFriday",
          "title": "Email",
          "url": "https://www.webbio.nl",
        },
        "facebook": Object {
          "nameSocialShare": "Facebook",
          "title": "BlackFriday",
          "url": "https://www.webbio.nl",
        },
        "twitter": Object {
          "hashtags": Array [
            "BlackFriday",
            "Netherlands",
            "Sale",
            "Shops",
          ],
          "nameSocialShare": "Twitter",
          "title": "BlackFriday",
          "url": "https://www.webbio.nl",
          "via": "BlackFriday",
        },
        "whatsapp": Object {
          "nameSocialShare": "WhatsApp",
          "separator": " - ",
          "title": "BlackFriday",
          "url": "https://www.webbio.nl",
        },
      },
      "smallBackLink": Object {
        "title": "Alle winkels",
        "url": "/",
      },
      "storeLink": Object {
        "title": "Beddenreus",
        "url": "/stores-single",
      },
      "timeLeftBar": Object {
        "text": "NU geldig",
        "value": 80,
      },
      "title": "15% korting op het hele assortiment",
    }
  }
/>
`;
